meilisearch-core/src/update/settings_update.rs

```diff
@@ -102,6 +102,19 @@ pub fn apply_settings_update(
         UpdateState::Nothing => (),
     }
 
+    // TODO smarter reindexing
+    match settings.attributes_for_faceting {
+        UpdateState::Update(attrs) => {
+            apply_attributes_for_faceting_update(writer, index, &mut schema, &attrs)?;
+            must_reindex = true;
+        },
+        UpdateState::Clear => {
+            apply_attributes_for_faceting_update(writer, index, &mut schema, &[])?;
```

Comment:
Does a clear must set an empty array here? Is the first ever written value an empty array or the value doesn't exist at all?

Reply:
no, we can indeed just delete the entry